src/document/transformoperation.js

```diff
@@ -0,0 +1,434 @@
+/**
```

Comment:
I think we could move transformoperation to the operation namespace. It is not an operation, but I think that the namespace is something more than the place only for operations. It could be called transform, so it will not be confused with the operations, so: `document/transformoperation` -> `document/operation/transform`